lib/puppet/type/file.rb

```diff
@@ -396,6 +396,10 @@ def self.title_patterns
     end
     # if the resource is a link, make sure the target is created first
     req << self[:target] if self[:target]
+    # if the resource has a source set, make sure it is created first
+    self[:source]&.each do |src|
+      req << src.delete_prefix('file://') if src.start_with?('file://')
```

Comment:
`delete_prefix` is Ruby 2.5+, which should be fine for Puppet 7.

I don't recall which Ruby Puppet 6 uses by default.

Reply:
In the tests below, none of them start with `file://` explicitly. Should this handle both `/blah` and `file:///blah`? Or am I missing a nuance?

Reply:
not using a schema prefix, but just an absolute file path is a shortcut for file:// and gets automatically translated internally.

https://puppet.com/docs/puppet/7/types/file.html#file-attribute-source:
> Fully qualified paths to locally available files (including files on NFS shares or Windows mapped drives).
> file: URIs, which behave the same as local file paths.

I can add an explicit test tho.